scripts/check-versions.py

```diff
@@ -0,0 +1,269 @@
+__license__ = "GPLv3"
+
+"""
+check-version.py is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+check-version.py is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+See <https://www.gnu.org/licenses/gpl-3.0.html> for full license text.
+"""
+
+import os
+import urllib.request
+import sys
+import re
+import hashlib
+from distutils.version import LooseVersion
+import yaml
+
+BASE_DIR = "../recipes-ros"
+DEBUG = True
+EXCLUDE = ["packagegroups"]
+DISTRO = "indigo"
+DIST_FILE = "https://raw.githubusercontent.com/ros/rosdistro/master/" + DISTRO+ "/distribution.yaml"
+dist_raw = None
+
+
+class MoveRepoExcetion(Exception):
+    pass
+
+def print_debug(text):
+    pass
+
+def print_err(text):
+    pre = '\033[91m'
+    post = '\033[0m'
+    print(pre + text + post)
+
+
+def print_ok(text):
+    pre = '\033[92m'
+    post = '\033[0m'
+    print(pre + text + post)
+
+
+def check_version(package, silent=False, details=False):
+    global dist_raw
+    printlist = []
+    if dist_raw is None:
+        dist_raw = yaml.load(urllib.request.urlopen(DIST_FILE).read())
+    package_dir = os.path.join(BASE_DIR, package)
+    versions = []
+    try:
+        dist_ver = dist_raw["repositories"][package.replace("-", "_")]["release"]["version"]
+        dist_ver = dist_ver.split("-")[0]
+    except KeyError:
+        dist_ver = ""
+    for filename in os.listdir(package_dir):
+        if filename.endswith(".bb"):
+            version = filename.split("_")[1].split(".bb")[0]
+            if version not in versions:
+                versions.append(version)
+
+            if not silent and details:
+                printlist.append(" - " + os.path.basename(filename))
+    if len(versions) > 1:
+        print("Package: %s" % package)
+        print(versions)
+        raise Exception("Multiple versions per package not supported at this time")
+
+    match = dist_ver == version
+    if dist_ver == "":
+        pre = post = ""
+        mid = "   "
+    elif version == "git":
+        mid = "   "
+        pre = post = ""
+    else:
+        pre = '\033[92m' if match else '\033[91m'
+        post = '\033[0m'
+        if match:
+            mid = " = "
+        else:
+            assert LooseVersion(version) != LooseVersion(dist_ver)
+            if LooseVersion(version) > LooseVersion(dist_ver):
+                mid = " > "
+            elif LooseVersion(dist_ver) > LooseVersion(version):
+                mid = " < "
+
+    if not silent:
+        print(package.ljust(35) + pre + version.ljust(10) + mid + dist_ver.ljust(10) + post)
+        for line in printlist:
+            print(line)
+    return match, version, dist_ver
```

Comment:
`version` seems to be unused anywhere.

Reply:
version is used in function `update_package`